From 3921290f1f76781189c79475bdeb2d2385999270 Mon Sep 17 00:00:00 2001
From: Gu Zitao <guzitao@wxiat.com>
Date: Tue, 22 Feb 2022 22:12:43 +0800
Subject: [PATCH] sw64: add AUDIT_ARCH_SW64 definition
Patch-mainline: NO, OTHERS
References: OLK-5.10
Modified-by-SEL: Yes, modified due to different context


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4SPZD
CVE: NA

-------------------------------

Add AUDIT_ARCH_SW64 definition to support syscall_get_arch(), which is
required to be implemented on all architectures in order to extend the
generic ptrace API with PTRACE_GET_SYSCALL_INFO request.

Signed-off-by: Gu Zitao <guzitao@wxiat.com> #openEuler_contributor
Signed-off-by: Laibin Qiu <qiulaibin@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/uapi/linux/audit.h  |    1 +
 include/uapi/linux/elf-em.h |    1 +
 2 files changed, 2 insertions(+)

--- a/include/uapi/linux/audit.h
+++ b/include/uapi/linux/audit.h
@@ -436,6 +436,7 @@ enum {
 #define AUDIT_ARCH_XTENSA	(EM_XTENSA)
 #define AUDIT_ARCH_LOONGARCH32	(EM_LOONGARCH|__AUDIT_ARCH_LE)
 #define AUDIT_ARCH_LOONGARCH64	(EM_LOONGARCH|__AUDIT_ARCH_64BIT|__AUDIT_ARCH_LE)
+#define AUDIT_ARCH_SW64		(EM_SW64|__AUDIT_ARCH_64BIT|__AUDIT_ARCH_LE)
 
 #define AUDIT_PERM_EXEC		1
 #define AUDIT_PERM_WRITE	2
--- a/include/uapi/linux/elf-em.h
+++ b/include/uapi/linux/elf-em.h
@@ -59,6 +59,7 @@
  * up with a final number.
  */
 #define EM_ALPHA	0x9026
+#define EM_SW64		0x9916
 
 /* Bogus old m32r magic number, used by old tools. */
 #define EM_CYGNUS_M32R	0x9041
